### DISCUSSION TITLE
How to deploy by docker as selfhost?

### DISCUSSION THREAD

**Discussion by mickey3721** [xiangechen/chili3d#54] [Q&A] [answered]
I want to deploy the 3d explorer in a selfhost server by docker, how to do?

**Comment by xiangechen** [comment, marked as answer]
Assuming you are already familiar with docker, this is a static page that can be deployed via nginx. The contents of the Dockerfile file in the project root directory are as follows:

```
FROM node:alpine AS builder
WORKDIR '/app'
COPY . .
RUN npm install && npm run build
 
FROM nginx:alpine
COPY --from=builder /app/dist /usr/share/nginx/html
```

**Comment by herobrinor** [comment]
In the last line of this Dockerfile, I changed /app/build to /app/dist and that works for me.

**Comment by mickey3721** [comment]
Did not build image successfully, both /app/build and /app/dist
![image](https://github.com/user-attachments/assets/3f6d59f7-865c-4a11-b026-3db650a53194)

can support docker image for deployment?

**Comment by xiangechen** [comment]
I've revised the previous comment; it should be okay now.
